### PR TITLE
chore: attempt to support postgresql ssl

### LINKED ISSUE #572: CommonServer does not support sslsecured databases.
https://github.com/OneUptime/oneuptime/blob/master/CommonServer/Infrastructure/PostgresConfig.ts#L16
https://github.com/OneUptime/oneuptime/blob/master/CommonServer/Infrastructure/PostgresDatabase.ts#L38

Connecting to a managed postgres database provider requires you to have sslmode set to require.
The current code does not allow for it to be set.

See [data-source-options for typeorm](https://typeorm.io/data-source-options#postgres--cockroachdb-data-source-options).

Proposed change would be to check to see if `process.env.DATABASE_CERT` or `process.env.DATABASE_KEY` or `process.env.DATABASE_CA` and set it in the ssl object.

### PR DESCRIPTION
**Pull request by Kashalls**
### Pull Request Checklist: 

- [x] Please make sure all jobs pass before requesting a review. 
- [x] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).
- [x] Have you lint your code locally before submission?
- [ ] Did you write tests where appropiate?

### Related Issue?
closes #572

### CODE REVIEW

File: CommonServer/Config.ts [hunk 1]
```diff
@@ -36,6 +36,22 @@ export const DatabasePassword: string =
...
+export const DatabaseSslKey: string | undefined =
+    process.env['DATABASE_SSL_KEY'] || undefined;
+
+export const DatabaseSslCert: string | undefined =
+    process.env['DATABASE_SSL_CERT'] || undefined;
+
+export const DatabaseRejectUnauthorized: boolean =
+    Boolean(process.env['DATABASE_SSL_REJECT_UNAUTHORIZED']) || false;
+
+export const shouldDatabaseSslEnable: boolean = Boolean(
```
Comment (simlarsen) [inline CommonServer/Config.ts:51]: Can this be pascal cased like others?

File: CommonServer/Infrastructure/PostgresConfig.ts [hunk 2]
```diff
@@ -23,6 +28,14 @@ export const dataSourceOptions: DataSourceOptions = {
...
     migrations: Migrations,
     entities: Entities,
+    ssl: shouldDatabaseSslEnable
+        ? {
+              rejectUnauthorized: DatabaseRejectUnauthorized,
+              ca: DatabaseSslCa,
+              key: DatabaseSslKey,
+              cert: DatabaseSslCert,
+          }
+        : false,
```
Comment (simlarsen) [inline CommonServer/Infrastructure/PostgresConfig.ts:38]: : undefined here?

**Review by simlarsen — CHANGES_REQUESTED** [review]

File: CommonServer/Config.ts [hunk 3]
```diff
@@ -36,6 +36,22 @@ export const DatabasePassword: string =
 export const DatabaseName: string =
     process.env['DATABASE_NAME'] || 'oneuptimedb';
 
+export const DatabaseSslCa: string | undefined =
+    process.env['DATABASE_SSL_CA'] || undefined;
```
Comment (simlarsen) [inline CommonServer/Config.ts:40]: Can you please also add these to config.example.env as empty strings and add comments around it?

**Review by simlarsen — CHANGES_REQUESTED** [review]

File: CommonServer/Config.ts [hunk 4]
```diff
@@ -36,6 +36,22 @@ export const DatabasePassword: string =
...
+    process.env['DATABASE_SSL_KEY'] || undefined;
+
+export const DatabaseSslCert: string | undefined =
+    process.env['DATABASE_SSL_CERT'] || undefined;
+
+export const DatabaseRejectUnauthorized: boolean =
+    Boolean(process.env['DATABASE_SSL_REJECT_UNAUTHORIZED']) || false;
+
+export const shouldDatabaseSslEnable: boolean = Boolean(
+    DatabaseSslCa || DatabaseSslCert || DatabaseSslKey
```
Comment (simlarsen) [inline CommonServer/Config.ts:52]: DatabaseSslCa || DatabaseSslCert || DatabaseSslKey -> should be  DatabaseSslCa && DatabaseSslCert && DatabaseSslKey

**Review by simlarsen — CHANGES_REQUESTED** [review]

**Comment by simlarsen** [comment]
Also, if those env vars are provided. It is possible to make postgres container over SSL?

File: CommonServer/Config.ts (hunk 4 above)
Comment (Kashalls) [reply CommonServer/Config.ts:52]: No, technically it should be added only if it's necessary.

**Comment by Kashalls** [comment]
> Also, if those env vars are provided. It is possible to make postgres container over SSL?

It requires the postgres container to have a valid tls chain + a fqdn

File: CommonServer/Infrastructure/PostgresConfig.ts (hunk 2 above)
Comment (Kashalls) [reply CommonServer/Infrastructure/PostgresConfig.ts:38]: https://github.com/typeorm/typeorm/blob/ff6e8751d98dfe9999ee21906cca7d20b1c6b15d/src/driver/postgres/PostgresConnectionCredentialsOptions.ts#L40

Typeorm expects a boolean false or TlsOptions

File: CommonServer/Config.ts (hunk 4 above)
Comment (Kashalls) [reply CommonServer/Config.ts:52]: In some instances where a valid Ca is provided, Cert and Key should not be provided. See: https://node-postgres.com/features/ssl

File: CommonServer/Config.ts (hunk 1 above)
Comment (Kashalls) [reply CommonServer/Config.ts:51]: I was trying to follow the formatting that was already prexisting: https://github.com/OneUptime/oneuptime/blob/master/CommonServer/Config.ts#L51

Comment (simlarsen) [reply CommonServer/Config.ts:51]: Ah ,looks like we'll have to pascal case that as well.

File: CommonServer/Infrastructure/PostgresConfig.ts (hunk 2 above)
Comment (simlarsen) [reply CommonServer/Infrastructure/PostgresConfig.ts:38]: Got it!

File: CommonServer/Config.ts (hunk 4 above)
Comment (simlarsen) [reply CommonServer/Config.ts:52]: So, DatabaseSslCa || (DatabaseSslCert && DatabaseSslKey) ?

File: CommonServer/Config.ts [hunk 5]
```diff
@@ -36,6 +36,22 @@ export const DatabasePassword: string =
...
+    process.env['DATABASE_SSL_CA'] || undefined;
+
+export const DatabaseSslKey: string | undefined =
+    process.env['DATABASE_SSL_KEY'] || undefined;
+
+export const DatabaseSslCert: string | undefined =
+    process.env['DATABASE_SSL_CERT'] || undefined;
+
+export const DatabaseRejectUnauthorized: boolean =
+    Boolean(process.env['DATABASE_SSL_REJECT_UNAUTHORIZED']) || false;
```
Comment (simlarsen) [inline CommonServer/Config.ts:49]: this should be process.env['DATABASE_SSL_REJECT_UNAUTHORIZED'] === "true". 

Boolean("false") is true

**Review by simlarsen — CHANGES_REQUESTED** [review]

**Review by simlarsen — APPROVED** [review]

**Pull request merged**